icu4c/source/tools/toolutil/json-json.hpp

```diff
@@ -7757,22 +7757,22 @@ class lexer : public lexer_base<BasicJsonType>
                                     }
                                     else
                                     {
-                                        error_message = "invalid string: surrogate U+D800..U+DBFF must be followed by U+DC00..U+DFFF";
-                                        return token_type::parse_error;
+                                        // ICU PATCH
+                                        codepoint = codepoint1;
```

Comment:
This should be upstreamed as an option.
Also, ICU PATCH here should probably link to a future ticket noting the patch is here, `// TODO ICU-nnnn Patched because …`

Reply:
I'm not sure if it would be accepted upstream; see https://json.nlohmann.me/home/faq/#parse-errors-reading-non-ascii-characters --

"Only UTF-8 encoded input is supported which is the default encoding for JSON according to [RFC 8259](https://tools.ietf.org/html/rfc8259.html#section-8.1)."

Since the codepoint in the test .json file is escaped, I'm not sure why it's considered to be invalid UTF-8, but maybe I'm not understanding something.